sawtooth-daml-rpc/src/main/java/com/blockchaintp/sawtooth/daml/rpc/DamlAuthServices.java

```diff
@@ -0,0 +1,166 @@
+import com.digitalasset.ledger.api.auth.ClaimPublic$;
+import com.digitalasset.ledger.api.auth.Claims;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.spongycastle.util.Arrays;
+
+import akka.protobuf.ByteString;
+import io.grpc.Metadata;
+import scala.collection.immutable.List;
+import scala.collection.immutable.List$;
+import scala.collection.mutable.ListBuffer;
+
+/**
+ * Responsible for decoding JWTToken sent from GRPC
+ *
+ */
+public class DamlAuthServices implements AuthService {
+
+  private final Algorithm ecdsa512Algorithm;
+  private ECPublicKey publicKey;
+
+  public DamlAuthServices(String pubKeyInHex) {
+    byte[] pubKey = ByteString.copyFromUtf8(pubKeyInHex).toByteArray();
+    int midPoint = pubKey.length / 2;
+    byte[] pubKeyX = Arrays.copyOfRange(pubKey, 0, midPoint);
+    byte[] pubKeyY = Arrays.copyOfRange(pubKey, midPoint, pubKey.length);
+    Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
+    ECPoint publicPoint = new ECPoint(new BigInteger(1, pubKeyX), new BigInteger(1, pubKeyY));
+    try {
+      AlgorithmParameters parameters = AlgorithmParameters.getInstance("EC");
+      parameters.init(new ECGenParameterSpec("secp256k1"));
+
+      ECParameterSpec params = parameters.getParameterSpec(ECParameterSpec.class);
+      ECPublicKeySpec pubKeySpec = new ECPublicKeySpec(publicPoint, params);
+
+      KeyFactory kf = KeyFactory.getInstance("EC");
+      this.publicKey = (ECPublicKey) kf.generatePublic(pubKeySpec);
+
+      this.ecdsa512Algorithm = Algorithm.ECDSA512(this.publicKey, null);
+    } catch (InvalidParameterSpecException | NoSuchAlgorithmException | InvalidKeySpecException nse) {
+      throw new RuntimeException(nse);
+    }
+
+  }
+
+  public final CompletionStage<Claims> decodeMetadata(final io.grpc.Metadata headers) {
+    try {
+      return CompletableFuture.completedFuture(decodeAndParse(headers));
+    } catch (final Exception e) {
+      return CompletableFuture.completedFuture(Claims.empty());
+    }
+  }
+
+  private com.digitalasset.ledger.api.auth.Claims decodeAndParse(final io.grpc.Metadata headers) throws Exception {
+
+    final String regex = "Bearer (.*)";
+    final Pattern pattern = Pattern.compile(regex);
+
+    final Metadata.Key<String> authorizationKey = Metadata.Key.of("Authorization", Metadata.ASCII_STRING_MARSHALLER);
+    final String authKeyString = headers.get(authorizationKey);
+    final Matcher matcher = pattern.matcher(authKeyString);
+    final String tokenString = matcher.group(1);
+    if (tokenString == null) {
+      throw new Exception();
+    }
+
+    final JWTVerifier verifier = JWT.require(this.ecdsa512Algorithm).build();
+    final DecodedJWT decodedJWT = verifier.verify(tokenString);
+    final AuthServiceJWTPayload jwtPayload = parsePayload(decodedJWT);
+    return payloadToDAClaims(jwtPayload);
+  }
+
+  private AuthServiceJWTPayload parsePayload(final DecodedJWT decodedJWT) {
```

Comment:
Is the payload any different to the standard one in here?
https://github.com/digital-asset/daml/blob/master/ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/AuthServiceJWTPayload.scala

Reply:
If the answer is "they are the same" it would really pay off to use the standard implementation and only merge the EC signature algorithm upstream

Reply:
@scealiontach your views, please?

Reply:
Believe you are using that one already, based on the imports.